src/radical/pilot/agent/radical-pilot-agent-multicore.py

```diff
@@ -2311,64 +2310,59 @@ def _configure(self):
         raise NotImplementedError("_Configure not implemented for LRMS type: %s." % self.name)
 
 
-    # --------------------------------------------------------------------------
-    #
-    @staticmethod
-    def hostname(logger=None):
-
-        hostname = socket.getfqdn()
-
-        if not hostname:
-            hostname = os.uname()[1]
-
-        if not hostname:
-            if logger:
-                logger.error("could not detect hostname")
-            raise RuntimeError("could not detect hostname")
-
-        return hostname
-
 
     # --------------------------------------------------------------------------
     #
     @staticmethod
-    def hostip(host=None, logger=None):
+    def hostip(req=None, logger=None):
         """
-        look up the ip number for a given host name.  If hostname is not given,
-        look up IP for localhost.
+        Look up the ip number for a given requested interface name.
+        If interface is not given, do some magic.
         """
 
-        if not host:
+        # List of interfaces that we probably dont want to bind to
+        black_list = ['lo', 'sit0']
 
-            # try the simple and safe detection first -- which will fail though
-            # if we don't have any connectivity
-            try:
-                s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
-                s.setblocking(False)
-                s.settimeout(1.0)
-                s.connect(('8.8.8.8', 53))
-                return s.getsockname()[0]
-            except Exception as e:
-                if logger:
-                    logger.exception("use fallback IP detection (%s)" % e)
+        # Known intefaces in preferred order
+        sorted_preferred = ['ipogif0', 'eth0']
```

Comment:
I like them if there are no alternatives.  In this case, I like 'ipogif0' as config entry even better :)  Is that hard-coded heuristics then still useful (ie. does it apply on any other host)?